### PR TITLE
[doc] feat: add Claude Code skills for add-dataset, add-reward, add-trainer

### PR DESCRIPTION
**Pull request by khazic**
### What does this PR do?

Adds three Claude Code skills for the most common veRL contribution patterns. Split out from #5843 per review from @tongyx361.

Each file under `.agents/skills/` is read by Claude Code when the user invokes the corresponding slash command. No runtime behavior is affected.

| Skill | Slash command | Purpose |
|---|---|---|
| `add-dataset` | `/add-dataset` | Preprocess + integrate a new RL training dataset |
| `add-reward` | `/add-reward` | Implement a `compute_score` reward function |
| `add-trainer` | `/add-trainer` | Add a new algorithm / trainer recipe |

---

#### `/add-dataset` — adding AQuA-RAT

**Prompt:** `/add-dataset` I want to add the `openai/aqua_rat` multiple-choice math dataset.

**Claude generates** `examples/data_preprocess/aqua_rat.py`:

```python
data_source = "openai/aqua_rat"

def make_map_fn(split: str):
    def process_fn(example: dict, idx: int) -> dict:
        options_str = "\n".join(example["options"])
        content = f"{example['question']}\n\nOptions:\n{options_str}"
        return {
            "data_source": data_source,
            "prompt": [
                {"role": "system", "content": "Think step by step, then select the correct option (A-E)."},
                {"role": "user", "content": content},
            ],
            "ability": "math",
            "reward_model": {"style": "rule", "ground_truth": example["correct"]},
            "extra_info": {"split": split, "index": idx},
        }
    return process_fn
```

The skill correctly applied the required schema (`data_source`, `prompt`, `reward_model.ground_truth`) and matched the dataset field names.

---

#### `/add-reward` — multiple-choice answer extraction

**Prompt:** `/add-reward` for `openai/aqua_rat` — extract the chosen option letter (A–E).

**Claude generates** `verl/utils/reward_score/aqua_rat.py`:

```python
import re

def compute_score(solution_str: str, ground_truth: str) -> float:
    matches = re.findall(r'\b([A-E])\b', solution_str.split("assistant")[-1])
    if matches and matches[-1] == ground_truth.strip().upper():
        return 1.0
    return 0.0
```

And registers it in `verl/utils/reward_score/__init__.py`:

```python
elif data_source == "openai/aqua_rat":
    from . import aqua_rat
    res = aqua_rat.compute_score(solution_str, ground_truth)
```

The skill correctly followed the no-exceptions, return-float contract and matched the `data_source` key from the preprocessing step.

---

#### `/add-trainer` — GRPO with clipped advantages

**Prompt:** `/add-trainer` I want a GRPO variant that clips advantages to `[-clip, clip]` before the policy update.

**Claude generates** `examples/grpo_clip_trainer/grpo_clip_trainer.py`:

```python
from verl.trainer.ppo.ray_trainer import RayPPOTrainer
from verl.trainer.ppo.core_algos import register_adv_est

@register_adv_est("grpo_clip")
def grpo_clip_estimator(token_level_rewards, response_mask, config, **kwargs):
    advantages, returns = _grpo_base(token_level_rewards, response_mask, config)
    clip = config.algorithm.get("advantage_clip", 5.0)
    advantages = advantages.clamp(-clip, clip)
    return advantages, returns

class GRPOClipTrainer(RayPPOTrainer):
    pass  # uses grpo_clip via config: algorithm.adv_estimator=grpo_clip
```

The skill correctly identified `register_adv_est` as the extension point and showed how to wire the config key through.

---

### Checklist

- [x] No runtime changes — `.agents/skills/` files only
- [x] Splits out part of #5843
- [x] Usage examples verified against actual veRL code structure

### Related
- Remaining skills will follow in separate PRs: `add-unit-tests`, `review-pr`, `create-pr`, `commit-conventions`, `debug-distributed`, upgrade skills

### CODE REVIEW

**Comment by khazic** [comment]
## Summary

This PR adds three Claude Code skills that guide AI assistants through the most common veRL contribution patterns.

**What each skill does:**
-  — walks through creating a preprocessing script with the correct parquet schema, wiring , and pointing the trainer config at the output files
-  — covers implementing a  function and registering it in 
-  — explains the  extension point for new advantage estimators, and when to subclass  for structural changes instead

**Improvements in the latest commit (based on Gemini review):**
- Corrected class name  →  to match the actual implementation
- Fixed  template signature to include , preventing a  at runtime
- Replaced the nonexistent  override with the correct  decorator pattern
- Fixed the run script entry point — the standard  does not load external modules automatically, so the custom estimator module must be imported first to trigger registration

File: .agents/skills/add-reward/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
```
Comment (ETOgaosion) [inline .agents/skills/add-reward/SKILL.md:1]: Now we switch to RewardLoop abstraction, may need updates this skill?

**Comment by khazic** [comment]
@ETOgaosion yes, it needed updating. Done in 68dc35fa.

The skill now targets Reward Loop instead of `verl/workers/reward_manager`: `compute_score` may be sync or async (the worker awaits async ones directly instead of holding an executor thread), custom managers subclass `RewardManagerBase` and implement `async run_single` returning `{"reward_score", "reward_extra_info"}` against the `verl.experimental.reward_loop.reward_manager` registry, and the config keys moved from `reward_model.reward_manager=...` to the `reward` group (`reward.reward_manager.name`, `reward.num_workers`, `reward.custom_reward_function.path`). The reference table lists naive / dapo / limited / remote. The legacy layer is mentioned once, as backward compatibility that new code should not target.

Three stale details the earlier bot review flagged are fixed in the same commit: `solution_str` is the decoded response only (both managers pass `response_str`), the `default_compute_score` snippet no longer reprints a truncated signature that would drop `sandbox_fusion_url` / `concurrent_semaphore` / `**kwargs`, and the dataset skill now says `RLHFDataset`.

File: .agents/skills/add-reward/SKILL.md (hunk 1 above)
Comment (khazic) [reply .agents/skills/add-reward/SKILL.md:1]: Yes, thanks for the pointer. The skill now targets Reward Loop end to end: compute_score (sync or async) plus a RewardManagerBase.run_single manager under verl/experimental/reward_loop, wired through the reward config group (reward.num_workers, reward.reward_manager.name, or source=importlib for an out-of-tree manager). The branch is also merged up to current main.

**Review by ETOgaosion — DISMISSED** [review]

File: .agents/skills/add-dataset/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+- User wants to preprocess a new dataset for GRPO/PPO training
+- User mentions creating a data preprocessing script
+
+## Overview
+
+veRL datasets follow a two-step pattern:
+
+1. **Preprocessing script** (`examples/data_preprocess/<name>.py`) — run once offline to
+   convert raw data into parquet files with a fixed schema
+2. **`RLDataset`** (`verl/utils/dataset/rl_dataset.py`) — runtime dataset class that
```
[the remark this answers is not in the document]
Comment (khazic) [reply .agents/skills/add-dataset/SKILL.md]: The text already refers to RLHFDataset; this thread predates that fix.

File: .agents/skills/add-reward/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+```
+
+### Step 2: Register in default_compute_score
+
+Update `verl/utils/reward_score/__init__.py` to include your new data source:
+
+```python
+from verl.utils.reward_score.<name> import compute_score as <name>_compute_score
+
+def default_compute_score(data_source, solution_str, ground_truth, extra_info=None):
```
[the remark this answers is not in the document]
Comment (khazic) [reply .agents/skills/add-reward/SKILL.md]: Kept **kwargs on the compute_score template and documented why: the reward manager forwards reward_router_address and reward_model_tokenizer when a router is configured.

File: .agents/skills/add-trainer/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+class MyTrainer(RayPPOTrainer):
+    """My custom RL trainer."""
+
+    def _compute_advantage(self, data: DataProto) -> DataProto:
+        """Override advantage computation for your algorithm."""
+        rewards = data.batch["token_level_scores"]  # shape: [bs, seqlen]
+        # ... your advantage computation
+        data.batch["advantages"] = advantages
+        data.batch["returns"] = returns
+        return data
```
[the remark this answers is not in the document]
Comment (khazic) [reply .agents/skills/add-trainer/SKILL.md]: Advantage customization now points at register_adv_est, with an explicit note that compute_advantage is a standalone function rather than a method.

File: .agents/skills/add-trainer/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+### Step 4: Write Run Script
+
+```bash
+#!/bin/bash
+# examples/<name>_trainer/run_qwen2-7b.sh
+
+set -x
+
+python3 -m verl.trainer.main_ppo \
```
[the remark this answers is not in the document]
Comment (khazic) [reply .agents/skills/add-trainer/SKILL.md]: Added the custom entry point a RayPPOTrainer subclass needs (a TaskRunner subclass passed to run_ppo, following one_step_off_policy), and the Option B run script now calls that module instead of verl.trainer.main_ppo.
